subworkflows/local/consensus/main.nf

```diff
@@ -1,73 +1,81 @@
 #!/usr/bin/env nextflow
 
-include { FGBIO_COPYUMIFROMREADNAME         } from "${projectDir}/modules/nf-core/fgbio/copyumifromreadname/main.nf"
-include { FGBIO_CALLDUPLEXCONSENSUSREADS    } from "${projectDir}/modules/nf-core/fgbio/callduplexconsensusreads/main.nf"
-include { FGBIO_CALLMOLECULARCONSENSUSREADS } from "${projectDir}/modules/nf-core/fgbio/callmolecularconsensusreads/main.nf"
-include { FGBIO_COLLECTDUPLEXSEQMETRICS     } from "${projectDir}/modules/nf-core/fgbio/collectduplexseqmetrics/main.nf"
-include { FGBIO_FASTQTOBAM                  } from "${projectDir}/modules/nf-core/fgbio/fastqtobam/main.nf"
-include { FGBIO_FILTERCONSENSUSREADS        } from "${projectDir}/modules/nf-core/fgbio/filterconsensusreads/main.nf"
-include { FGBIO_GROUPREADSBYUMI             } from "${projectDir}/modules/nf-core/fgbio/groupreadsbyumi/main.nf"
-include { FGBIO_SORTBAM                     } from "${projectDir}/modules/nf-core/fgbio/sortbam/main.nf"
-include { FGBIO_ZIPPERBAMS                  } from "${projectDir}/modules/nf-core/fgbio/zipperbams/main.nf"
+include { FGBIO_COPYUMIFROMREADNAME         } from '../../../modules/nf-core/fgbio/copyumifromreadname/main'
+include { FGBIO_CALLMOLECULARCONSENSUSREADS } from '../../../modules/nf-core/fgbio/callmolecularconsensusreads/main'
+include { FGBIO_FASTQTOBAM as FASTQTOBAM_READNAME           } from '../../../modules/nf-core/fgbio/fastqtobam/main'
+include { FGBIO_FASTQTOBAM as FASTQTOBAM_SEQ           } from '../../../modules/nf-core/fgbio/fastqtobam/main'
+include { FGBIO_FILTERCONSENSUSREADS        } from '../../../modules/nf-core/fgbio/filterconsensusreads/main'
+include { FGBIO_GROUPREADSBYUMI             } from '../../../modules/nf-core/fgbio/groupreadsbyumi/main'
+include { FGBIO_SORTBAM                     } from '../../../modules/nf-core/fgbio/sortbam/main'
+include { FGBIO_ZIPPERBAMS                  } from '../../../modules/nf-core/fgbio/zipperbams/main'
+include { SAMTOOLS_FASTQ                    } from '../../../modules/nf-core/samtools/fastq/main'
+include { SAMTOOLS_INDEX                    } from '../../../modules/nf-core/samtools/index/main'
+include { BWA_MEM                        } from '../../../modules/nf-core/bwa/mem/main'
+include { samplesheetToList } from 'plugin/nf-schema'
+
 
 workflow CONSENSUS {
     take:
-        ch_fastq                // channel: tuple(meta, fastq1, fastq2) for SE/PE/duplex samples
-        ch_reference            // channel: reference genome fasta file
-        ch_umi_in_readname      // boolean
-
+        ch_input_fastq                   // channel: [meta_with_readgroup, fastq] for SE/PE/duplex samples
+        ch_genomes                       // map: reference genome files
+        ch_umi_in_readname               // boolean
```

Comment:
`ch_umi_in_readname` does not have to a channel (this makes your flow more complex than it should be)